test/e2e/far_e2e_test.go

```diff
@@ -201,17 +206,19 @@ func buildNodeParameters(clusterPlatformType configv1.PlatformType) (map[v1alpha
 	return testNodeParam, nil
 }
 
-// randomizeWorkerNode returns a worker node name which is different than the previous one
+// randomizeWorkerNode returns a worker node that his name is different than the previous one
```

Comment:
IIUC this returns a random node out of the given node list.

>that his name is different than the previous one

I don't understand this part

Reply:
Since we randomly pick a worker node and there are only 3 available nodes then we can pick the same node, even though we use a different seed for the random function. Therefore, we store the name of the last unhealthy node so we won't try to create second CR for the same node. 
> previous one

Refers to he previous remediation, so on the first remediation previousNodeName is nil, and on the second remediation it stores the name of the previous unheathy node in order to not create a CR on that node again.

Reply:
I see now.
I advise against such implementation.
- IMO it's bad practice to use global var in such way (not concurrent safe and confusing)
- I think the method is confusing
- we have no idea how many iteration will occur before the for condition is broker because we can always get randomly the same node, also seems like a risk of a deadlock.

Instead I suggest the following , let `randomizeWorkerNode` do just that - get a random node out of the list:
```
func randomizeWorkerNode(nodes *corev1.NodeList) *corev1.Node {
	// Generate a random seed based on the current time
	r := rand.New(rand.NewSource(time.Now().UnixNano()))
	// Randomly select a worker node
	return &nodes.Items[r.Intn(len(nodes.Items))]
}
```

And just pass to it an already filtered list (filter out the node you don't want it to get).